pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -94,15 +94,26 @@ func (p *LaunchTemplateProvider) Get(ctx context.Context, constraints *v1alpha1.
 		return nil, err
 	}
 	// Get constrained AMI ID
-	amis, err := p.amiProvider.Get(ctx, instanceTypes)
+	amis, err := p.amiProvider.Get(ctx, constraints, instanceTypes)
 	if err != nil {
 		return nil, err
 	}
 	// Construct launch templates
 	launchTemplates := map[string][]cloudprovider.InstanceType{}
 	for amiID, instanceTypes := range amis {
 		// Get userData for Node
-		userData, err := p.getUserData(ctx, constraints, instanceTypes, additionalLabels)
+		var userData string
+		if constraints.AMIFamily != nil {
+			if *constraints.AMIFamily == v1alpha1.OperatingSystemBottleRocket {
```

Comment:
can you use `if strings.EqualFold(*constraints.AMIFaily, v1alpha1.OperatingSystemBottlerocket)` on both of these conditionals for a case-insensitive equals?

Reply:
TIL!